mixer/adapter/fluentd/fluentd.go

```diff
@@ -144,15 +224,35 @@ func (h *handler) HandleLogEntry(ctx context.Context, insts []*logentry.Instance
 			delete(i.Variables, "tag")
 		}
 
-		if err := h.logger.PostWithTime(tag.(string), i.Timestamp, i.Variables); err != nil {
-			return err
+		data, err := h.logger.EncodeData(tag.(string), i.Timestamp, i.Variables)
```

Comment:
Why encode and discard?  If we don't have room, we should discard at the earliest.

Can we run encode on the other side of the channel?

Reply:
Encoding beforehand gave a way to signal via metrics that a request was being dropped if it was too large.

In any case, I've moved the encoding to behind the channel.